cyberbrain/tracer.py

```diff
@@ -81,11 +100,16 @@ def _initialize_frame_and_logger(
             frame=self.frame,
             debug_mode=self.debug_mode,
         )
+        # Check if the traceback has been run, if so set the frame defined_lineno
```

Comment:
We can move this to `Frame`, and just need to have `self.defined_lineno = get_function_lineno_from_tracer()`.

Reply:
Or does it not work if called there?

Reply:
Yeah the stack summary doesn't hold the program state we'd want if you put it in the Frame unfortunately. The most consistent place to have it is in the Tracer's decorator function.

Reply:
Ok, then how about this? We still call `get_function_lineno_from_tracer()` within tracer, but store the result in `Frame.defined_lineno`. This way we can achieve the goal, as well as maintaining the right level of abstraction.

Reply:
I initially tried this but the frame wasn't initialised yet. So it was throwing an attribute error due to none type. This is if try self.frame.defined_lineno = get_function_lineno() in the tracers decorator. Any suggestions of how to get past that?

Reply:
Let's just store lineno in two places(tracer and frame) for now. I'll think about how to make it better.